packages/spanner/test/session-pool.js

```diff
@@ -1376,23 +1345,143 @@ describe('SessionPool', function() {
       });
     });
 
-    describe('listens for next opening', function() {
-      it('should call itself again when available is emitted', function(done) {
-        sessionPool.once = function(event, onAvailable) {
-          sessionPool.once = util.noop;
+    describe('polls for next opening', function() {
+      it('should poll for the next available session', function(done) {
+        sessionPool.pollForSession_ = function(callback) {
+          callback();
+        };
+
+        sessionPool.getNextAvailableSession_(OPTIONS, done);
+      });
+    });
+  });
 
-          assert.strictEqual(event, 'available');
+  describe('pollForSession_', function() {
+    var setInterval_;
 
-          assert.strictEqual(
-            onAvailable(),
-            sessionPool.getNextAvailableSession_(OPTIONS, assert.ifError)
-          );
+    beforeEach(function() {
+      setInterval_ = global.setInterval;
+      global.setInterval = function(cb) {
+        global.setInterval.calledWith_ = arguments;
+        setImmediate(cb);
+        return 1;
+      };
+    });
 
+    afterEach(function() {
+      global.setInterval = setInterval_;
+    });
+
+    it('should capture the acquire request', function() {
+      var callback = function() {};
+
+      sessionPool.pollForSession_(callback);
+      assert.strictEqual(sessionPool.pendingAcquires.length, 1);
+
+      var pendingAcquire = sessionPool.pendingAcquires[0];
+
+      assert.strictEqual(pendingAcquire.callback, callback);
+      assert.strictEqual(pendingAcquire.timeout, sessionPool.acquireTimeout);
+    });
+
+    it('should bail if it detects an acquire interval exists', function() {
+      var callback = function() {};
+
+      sessionPool.acquireInterval = 1;
+      sessionPool.pollForSession_(callback);
+
+      assert.strictEqual(global.setInterval.calledWith_, undefined);
+    });
+
+    it('should poll for a free session', function() {
+      var callback = function() {};
+
+      sessionPool.pollForSession_(callback);
+
+      var intervalArgs = global.setInterval.calledWith_;
+
+      assert.strictEqual(sessionPool.acquireInterval, 1);
+      assert.strictEqual(typeof intervalArgs[0], 'function');
+      assert.strictEqual(intervalArgs[1], 30000);
+    });
+
+    it('should call getNextAvailableSession_ if a read free', function(done) {
+      sessionPool.getNextAvailableSession_ = function(callback) {
+        callback();
```

Comment:
Can you leave a comment here and anywhere else `callback === done` that this is the value of `done`, typically `// done()`